### ISSUE TITLE
TM1Py performance after upgrade 1.11.3 >> 2.0.4

### ISSUE THREAD

**Issue by hermie64** [cubewise-code/tm1py#1176]
I noticed a performance decrease after upgrading TM1Py from version 1.11.3 to 2.0.4. 

I ran some tests in QA before upgrading our production environment, where I copied a process and a dimension.
With version 1.11.3, both operations took around 30 seconds, but with version 2.0.4, they take more than twice the time.

What could be causing this?

Best regards,

**Version**
- TM1py: 1.11.3 >> 2.0.4 
- TM1 Server Version: 11.8.02300.10

**Comment by MariusWirtz** [comment]
Thanks for the feedback @hermie64 

I assume you have exactly the same code in both setups, right?

Are you passing the same arguments to `TM1Service` in both setups?

Can you spot where exactly it takes the extra time? is it the authentication or the read or the write operation that has gotten slower?

**Comment by hermie64** [comment]
Yes, I used the same code and arguments in the same TM1 instance. I simply upgraded to version 2.0.4, noticed the performance decrease, then reinstalled version 1.11.3 and tested again.

I’ll try to gather more details to pinpoint exactly where the extra time is being spent.

Best regards,

**Comment by hermie64** [comment]
I tested both versions, 1.11.3 and 2.0.4, on my own workstation, and I don't observe the same performance issue.

However, on our servers, we kick off TM1Py processes via TI. As the next step, I added some logging statements to the Python code to gather more information.

I noticed that with version 1.11.3, there’s already a significant delay (about 30 seconds) before the first logging statement is written to the log file—this doesn’t happen on my workstation. With version 2.0.4, this delay is even longer, taking over 60 seconds. Interestingly, the performance of the Python/TM1Py code itself is more or less the same, regardless of the version.

Could this be related to something like a firewall issue?

**Comment by MariusWirtz** [comment]
Virus scanner / security software is the most likely explanation. 

Or is it possible that the CAM authentication takes such a long time? That could explain differences between runs? 

I'm just thinking out loud, but perhaps if you compile the script to exe with PyInstaller, it's handled differently?

I will move this to discussions as it's technically not a TM1py issue.
